src/main/ml-modules/root/server/endpoints/adhoc/wizard/api-adhoc-query-wizard-doc-selection.xqy

```diff
@@ -0,0 +1,138 @@
+at "/server/lib/const.xqy";
+import module namespace lib-adhoc = "http://marklogic.com/data-explore/lib/adhoc-lib" at "/server/lib/adhoc-lib.xqy";
+import module namespace lib-adhoc-create = "http://marklogic.com/data-explore/lib/adhoc-create-lib" at "/server/lib/adhoc-create-lib.xqy";
+import module namespace to-json = "http://marklogic.com/data-explore/lib/to-json" at "/server/lib/to-json-lib.xqy";
+import module namespace xu = "http://marklogic.com/data-explore/lib/xdmp-utils" at "/server/lib/xdmp-utils.xqy"; 
+import module namespace  check-user-lib = "http://www.marklogic.com/data-explore/lib/check-user-lib" at "/server/lib/check-user-lib.xqy" ;
+import module namespace detail-lib = "http://www.marklogic.com/data-explore/lib/detail-lib" at "/server/lib/detail-lib.xqy";
+
+declare function local:get-uris-by-directory($path, $max-uris,$start-uri,$db) {
+let $max-uris:=xs:int($max-uris)
+let $max-uris-plus1:=$max-uris+1
+let $path:=if(fn:ends-with($path,"/")) then $path else $path||"/"
+let $path:=if(fn:starts-with($path,"/")) then $path else "/"||$path
+
+let $query:=<query>
+cts:uris("{$start-uri}",('score-zero',"limit={$max-uris-plus1}"),cts:directory-query("{$path}","infinity"))[2 to {$max-uris-plus1}]
+</query>/text()
+    return xu:eval($query,(),
+    <options xmlns="xdmp:eval">
+      <database>{ xdmp:database($db) }</database>
+    </options>
+  )
+};
+declare function local:get-uris-by-collection-name($collection-name, $max-uris,$start-uri,$db) {
+   let $max-uris:=xs:int($max-uris)
+   let $max-uris-plus1:=$max-uris+1
+   let $query:=<query>cts:uris("{$start-uri}",('score-zero',"limit={$max-uris-plus1}"),
+                     cts:collection-query("{$collection-name}"))
+                     [2 to {$max-uris-plus1}]
+                  </query>/text()
+    
+    return xu:eval($query,(),
+    <options xmlns="xdmp:eval">
+      <database>{ xdmp:database($db) }</database>
+    </options>
+  )
+};
+declare function local:get-uris-by-root-element-name($element-name, $max-uris,$start-uri,$db) {
+    let $max-uris:=xs:int($max-uris)
+    let $max-uris-plus1:=$max-uris+1
+    let $query:=let $root-name:=tokenize($element-name,"~")[1]
+                let $ns:=tokenize($element-name,"~")[2]
+                return 
+                  <query>
+                     cts:uris("{$start-uri}",('score-zero',"limit={$max-uris-plus1}"),cts:element-query(fn:QName("{$ns}","{$root-name}"),cts:true-query()))
+                     [2 to {$max-uris-plus1}]
+                  </query>/text()
+    
+    return xu:eval($query,(),
+    <options xmlns="xdmp:eval">
+      <database>{ xdmp:database($db) }</database>
+    </options>
+  )
+};
+declare function local:get-root-element-names($db) {
+    let $query:=<query>
+                    declare variable $sampleSize := 384; (: 95% +/-5% confidence :)
+                    declare variable $sampleThreshold := 1000;
+                    let $numDocs := xdmp:estimate(cts:search(/, cts:and-query(())))
```

Comment:
Can do cts:true-query() instead of cts:and-query(()) (and likewise cts:false-query() instead of cts:or-query(()) )

Reply:
changed to cts:true-query()